### ISSUE TITLE
How to read obProperty.settings

### ISSUE THREAD

**Issue by amdad** [lovata/oc-shopaholic-plugin#37]
Hello, I'm trying to put in Twig Product template, property settings information to distinct between mediafinder mode image vs file. But array gives me null.  I'm doing something wrong, or $jsonable is missing in Model?

````twig
{# Get product item #}
{% set obProduct = ProductPage.get() %}
<div data-id="{{ obProduct.id }}">
    <h1>{{ obProduct.name }}</h1>
    {# Get list of proeprties for product #}
    {% set obPropertyList = obProduct.property %}
    {% if obPropertyList.isNotEmpty == true %}
        <ul>       
            {% for obProperty in obPropertyList if obProperty.hasValue() %}
                <li>
                
                 {% for setting in obProperty.settings  %}

{# HERE >>  #}   {{ setting.mediafinder }}
        
                 {% endfor %}
                
                {{ obProperty.type }}  {{ obProperty.name }}: {{ obProperty.property_value.getValueString() }}</li 
            {% endfor %}
        </ul>
    {% endif %}
</div>
````

**Comment by kharanenka** [comment]
Hi! We do not save the array of settings in the [PropertyItem](https://github.com/lovata/oc-shopaholic-plugin/wiki/PropertyItem) object to the cache. So now you get null. What task do you want to solve using the settings array? Maybe you can use the "code" field to solve your task? In our projects, we do not use this field in the twig templates, but if this is necessary, we can add the settings field in the next update.

**Comment by amdad** [comment]
mediafinder have 2 modes, image and file.
date have 3 modes, date, time and date & time.

Twig should have IFs for all this modes to show up different templates. For mediafinder image it will be <img src=""> tag, for file it will be <href url=""> download link. for dates time only, it can be clock, for date, calendar. When both calendar with time. More options we have  for a field more params we need to read. So please, add it to cache. 

If I can read: {{ obProperty.type }}  {{ obProperty.name }}, I need {{ obProperty.settings }} also.

I want to extend or add some fields in future so I would also rely more and more on settings field, because it's json and I can pack a lot of stuff there. It is or will be possible to add own types/modes/settings of custom fields?

**Comment by kharanenka** [comment]
Ok,  we add the settings field in the next update. We plan to develop our plugins and increase their functionality. If you have ideas for improving plugins, then you can create new issues. We will be happy your ideas.

**Comment by amdad** [comment]
Can you give me quick patch before update? Where I can fix it?
Yes, I will bring more ideas soon :)

**Comment by kharanenka** [comment]
You can add code in plugins/lovata/propertiesshopaholic/classes/item/PropertyItem.php file, line 97:
```php
...
    'measure_id'    => $this->obElement->measure_id,
    'settings'      => $this->obElement->settings,
];
...
```
Сlear the cache after adding the code

**Comment by amdad** [comment]
Thanks! :)

**Comment by amdad** [comment]
Quick question at the and of this Issue. (not related)
Why you do not include default templates in plugins components but only in docs? It's not finished yet or there is some other reason?

**Comment by kharanenka** [comment]
Each site has its own design and its HTML structure. We do not want to impose our HTML structure on developers. We want to give developers an easy and convenient tool for website development. Our components can be used for different tasks. For example: component [ProductList](https://github.com/lovata/oc-shopaholic-plugin/wiki/ProductList) can be used to display a catalog of products (a classic task), for display on the index page of a block with random products (custom block). The component structure allows developers to write less code for custom blocks. The components return the objects of the classes *Item and *Collection. These classes are easily expanded (see the extension of the [item](https://github.com/lovata/oc-toolbox-plugin/wiki/ElementItem#extending) and the [collection](https://github.com/lovata/oc-toolbox-plugin/wiki/ElementCollection#extending) for more details).

We are now preparing for release a demo-site where you can find ready-made templates. We plan to add a separate plugin in the future, in which there will be a set of templates. The demo-site code will be available for download and study

**Comment by amdad** [comment]
Default template of component not impose any code. But rather give us fully working functionality. 
Hold CTRL and double click on {% component 'Cart' %} and you have whole default code before you to edit your own much quicker then copy and paste from docs. Also, when you OK with staying with default (not custom) template it have significant advantage, because your website template is already updated when plugin add new feature in newer version. So we do not need check and update each view for new functionality. 

I very like you made components mixable and extendable with classes. For you wide view on topic I choose Shopaholic from the start and skip all the rest shop plugins from marketplace. But it really doesn't handicap default templates to be there is someone wants to use it. I don't see a reason why you want ship another plugin with default templates when you can add default templates to existing components because default component templates are completely optional in October. Docs can cover much more extensive examples. But still, current components cannot do anything by it's own without custom code for each of them, and need to be updated all the time by hand. First time in October plugin I see situation like this. So I don't understand this why you wouldn't change this situation? Again, default template only ADD ease of use to what you have here and doesn't take back anything from it. 

Im my own plugins I ship default templates to be most basic version of fully working plugin. Most of the times I use custom code over them. But most of scenarios are mixed. So some components are default ones and custom only where I need it. It reduce work after update to where it's really needed.

**Comment by kharanenka** [comment]
Thank you for your feedback. We will discuss your idea in the near future. You are right, simple templates in the components will be useful.

**Comment by amdad** [comment]
Yes, there are some projects that just need working functionality and custom theme css styling is enough. So default templates should be as simple and semantic as it could be. No bootstrap! :D To make it framework agnostic ;)

````twig
<div class="my-custom-cssclass-over-brandlist">
    {% component 'BrandList' %}
</div>

<div class="my-custom-cssclass-over-categorylist">
    {% component 'CategoryList' %}
</div>

<div class="my-custom-cssclass-over-productlist">
    {% component 'ProductList' %}
</div>

{# or #}

<div class="my-custom-shop-index">
    {% component 'BrandList' %}
    {% component 'CategoryList' %}
    {% component 'ProductList' %}
</div>
````

Go The October Way ;) Cheers!

**Comment by kharanenka** [comment]
Fixed in version 1.1.0 of "Properties for Shopaholic" plugin
